x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/e5_multilingual_callout/e5_multilingual_callout_data.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useEffect, useState } from 'react';
+
+import { useValues } from 'kea';
+
+import { IndexViewLogic } from '../../../index_view_logic';
+
+import { E5MultilingualCallOutProps, E5MultilingualCallOutState } from './e5_multilingual_callout';
+import { E5MultilingualCalloutLogic } from './e5_multilingual_callout_logic';
+
+export const E5_MULTILINGUAL_CALL_OUT_DISMISSED_KEY =
+  'enterprise-search-e5-multilingual-callout-dismissed';
+
+const isDismissed = () => localStorage.getItem(E5_MULTILINGUAL_CALL_OUT_DISMISSED_KEY) === 'true';
+
+export const useE5MultilingualCallOutData = ({
+  isCompact = false,
+  isDismissable = false,
+}: E5MultilingualCallOutProps): E5MultilingualCallOutState => {
+  const { ingestionMethod } = useValues(IndexViewLogic);
+  const { isCreateButtonDisabled, isModelRunningSingleThreaded, isStartButtonDisabled } = useValues(
+    E5MultilingualCalloutLogic
+  );
+
+  const [show, setShow] = useState<boolean>(() => {
+    if (!isDismissable) return true;
+
+    try {
+      return !isDismissed();
+    } catch {
+      return true;
+    }
+  });
+
+  useEffect(() => {
+    try {
+      if (!isDismissed()) {
+        localStorage.setItem(E5_MULTILINGUAL_CALL_OUT_DISMISSED_KEY, JSON.stringify(!show));
```

Comment:
@efegurkan 
I tried to move the localStorage to dismiss callback. Now, the callout is dismissed after clicking the dismiss button. However, the callout reappears after we refresh the whole page. Do you know how I can fix it?

```
const isDismissed = () => {
  try {
    return localStorage.getItem(E5_MULTILINGUAL_CALL_OUT_DISMISSED_KEY) === 'true';
  } catch {
    return false;
  }
};

export const useE5MultilingualCallOutData = ({
  isCompact = false,
  isDismissable = false,
}: E5MultilingualCallOutProps): E5MultilingualCallOutState => {
  const { ingestionMethod } = useValues(IndexViewLogic);
  const { isCreateButtonDisabled, isModelRunningSingleThreaded, isStartButtonDisabled } = useValues(
    E5MultilingualCalloutLogic
  );

  const [show, setShow] = useState<boolean>(() => {
    if (!isDismissable) return true;
    return !isDismissed();
  });

  const dismiss = useCallback(() => {
    setShow(false);
    localStorage.setItem(E5_MULTILINGUAL_CALL_OUT_DISMISSED_KEY, JSON.stringify(!show));
  }, []);

  return {
    dismiss,
    ingestionMethod,
    isCompact,
    isCreateButtonDisabled,
    isDismissable,
    isSingleThreaded: isModelRunningSingleThreaded,
    isStartButtonDisabled,
    show,
  };
};
```

Reply:
We have several examples of dismissable callouts in our frontend. This PR contains a concise example of how to do it using the existing useLocalStorage hook. You should be re-using that hook instead of creating your own.  https://github.com/elastic/kibana/pull/172147

Your second attempt isn't working because something's going wrong with what you're storing in localStorage. All of the negative boolean checks make it hard to figure out what's going on, but you're only ever reading from localStorage on initial load and after that it's all local state--so your local state is working, but your localStorage read/write interaction is not. Use the existing hook to fix that. 

Do this instead:
```
const [calloutDismissed, setCalloutDismissed] = useLocalStorage<boolean>(
    E5_MULTILINGUAL_CALL_OUT_DISMISSED_KEY,
    false
  );
```

Then instead of putting all of this logic into a hook that is too complicated, just directly use that hook in your component, and ignore its result where you don't want the component to be dismissable.